aws/data_source_aws_wafv2_regex_pattern_set.go

```diff
@@ -83,18 +84,18 @@ func dataSourceAwsWafv2RegexPatternSetRead(d *schema.ResourceData, meta interfac
 	}
 
 	resp, err := conn.GetRegexPatternSet(&wafv2.GetRegexPatternSetInput{
-		Id:    aws.String(*foundRegexPatternSet.Id),
-		Name:  aws.String(*foundRegexPatternSet.Name),
+		Id:    foundRegexPatternSet.Id,
+		Name:  foundRegexPatternSet.Name,
 		Scope: aws.String(d.Get("scope").(string)),
 	})
 
-	if err != nil {
+	if err != nil || resp == nil || resp.RegexPatternSet == nil {
```

Comment:
i'd recommend keeping the `err != nil` condition separate from the latter 2 (so we don't lose the err object on return), resulting in 2 `if` blocks